src/main/kotlin/org/rust/lang/core/psi/util/RustPsiNavigationExtensions.kt

```diff
@@ -146,12 +146,12 @@ private fun PsiElement.getTopmostParentInside(parent: PsiElement): PsiElement {
 
 private fun collectElements(start: PsiElement, stop: PsiElement?, pred: (PsiElement) -> Boolean): Array<out PsiElement> {
     val list = mutableListOf<PsiElement>()
```

Comment:
I think it's a good idea to add `check (stop == null || start.parent == stop.parent)` here.